core/src/test/java/org/apache/iceberg/TestTables.java

```diff
@@ -218,9 +222,7 @@ public OutputFile newOutputFile(String path) {
 
     @Override
     public void deleteFile(String path) {
-      if (!new File(path).delete()) {
-        throw new RuntimeIOException("Failed to delete file: " + path);
-      }
+      DELETED_FILE_PATHS.add(path);
```

Comment:
Instead of updating a static variable that we need to clean up between test cases, how about using the delete callback? I think that's going to be a better way to test.

Reply:
+1